static-web/search/search-box.js

```diff
@@ -726,25 +734,30 @@ class SearchBox {
             this.firstOption = null;
             this.lastOption = null;
             this.currentOption = null;
-            this.listboxNode.appendChild(this.noResultsElement);
+            this.listboxNode.textContent = "";
+            this.listboxNode.append(this.noResultsElement);
             return null;
         }
 
-        /**
-         * @type {SearchResult|null}
-         */
-        let newCurrentOption = null;
-
-        /** @type {(Fuzzysort.Result|TextMatch) []} */
-        let allResults = [];
+        let /** @type {(Fuzzysort.Result|TextMatch) []} */ allResults = [];
         allResults.push(...textResults);
         allResults.push(...results);
         allResults.sort((x, y) => y.score - x.score);
         allResults = allResults.slice(0, 30);
 
-        this.filteredOptions = [];
-        this.firstOption = null;
-        this.lastOption = null;
+        // The following computation is async, and so might fall behind later search results if
+        // the user types quickly. This counter detects whether the current function has become
+        // stale due to a later-starting search request
+        this.requestCounter += 1;
+        const requestCount = this.requestCounter;
```

Comment:
Add some comment here explaining that this increment is happening here for a reason.